### ISSUE TITLE
Do not crash without YAML-Header

### ISSUE THREAD

**Issue by cocker-cc** [smortex/yaml-sort#3]
With a YAML lacking the YAML-Header (tree Dashes) `yaml-sort` just crashes with Traceback, instead of giving an Error-Message, or just being error-tolerant.

**Comment by smortex** [comment]
I can't reproduce this with my example files, can you please provide symptomatic test cases so that I can use them as a starting point?

The START_OF_DOCUMENT / END_OF_DOCUMENT sucks and would benefit from some re-thinking:
https://github.com/smortex/yaml-sort/blob/c60d1aab94d04291244824620415f964d432e796/lib/yaml/sort/parser.ra#L10-L13

**Comment by cocker-cc** [comment]
```
$ cat tmp/a.yaml
a: AAA

$ yaml-sort tmp/a.yaml 
---
a: AAA
```
```
$ cat tmp/b.yaml
b:
- B
- BB

$ yaml-sort tmp/b.yaml 
Traceback (most recent call last):
        11: from /home/cocker/.rvm/gems/ruby-2.7.2/bin/yaml-sort:23:in `<main>'
        10: from /home/cocker/.rvm/gems/ruby-2.7.2/bin/yaml-sort:23:in `load'
         9: from /home/cocker/.rvm/gems/ruby-2.7.2/gems/yaml-sort-1.0.0/exe/yaml-sort:7:in `<top (required)>'
         8: from /home/cocker/.rvm/gems/ruby-2.7.2/gems/yaml-sort-1.0.0/lib/yaml/sort/cli.rb:45:in `execute'
         7: from /home/cocker/.rvm/gems/ruby-2.7.2/gems/yaml-sort-1.0.0/lib/yaml/sort/cli.rb:45:in `each'
         6: from /home/cocker/.rvm/gems/ruby-2.7.2/gems/yaml-sort-1.0.0/lib/yaml/sort/cli.rb:46:in `block in execute'
         5: from /home/cocker/.rvm/gems/ruby-2.7.2/gems/yaml-sort-1.0.0/lib/yaml/sort/cli.rb:55:in `process_document'
         4: from /home/cocker/.rvm/gems/ruby-2.7.2/gems/yaml-sort-1.0.0/lib/yaml/sort/cli.rb:68:in `sort_yaml'
         3: from parser.ra:176:in `parse'
         2: from (eval):3:in `do_parse'
         1: from (eval):3:in `_racc_do_parse_c'
parser.ra:180:in `on_error': undefined method `[]' for nil:NilClass (NoMethodError)
```

**Comment by smortex** [comment]
Thanks!

I just released 1.0.1 which should fix this.
